CONTRIBUTING.md

```diff
@@ -24,30 +24,70 @@ include those additions in the master version of this project, check out this
 ### Community
 
 This project is maintained by the [NeuroTechX](www.neurotechx.com) community. Join the
-[NeuroTechX Slack](https://docs.google.com/forms/d/e/1FAIpQLSfZyzhVdOLU8_oQ4NylHL8EFoKLIVmryGXA4u7HDsZpkTryvg/viewform),
-where discussions about MOABB takes place.
+[Gitter](https://gitter.im/moabb_dev/community), where discussions about MOABB takes
+place.
 
 ## How can I contribute?
 
-This is currently a small, humble project so our contribution process is rather casual. If
-there's a feature you'd be interested in building, go ahead! Let us know on the
-[NeuroTechX Slack](https://docs.google.com/forms/d/e/1FAIpQLSfZyzhVdOLU8_oQ4NylHL8EFoKLIVmryGXA4u7HDsZpkTryvg/viewform)
-or [open an issue](../../issues) so others can follow along and we'll support you as much
-as we can. When you're finished submit a pull request to the master branch referencing the
-specific issue you addressed.
-
-If you find a bug, or have a suggestion on how to improve the project, just fill out a
-[Github issue](../../issues)
+If there's a feature you'd be interested in building or you find a bug or have a
+suggestion on how to improve the project, go ahead! Let us know on the
+[Gitter](https://gitter.im/moabb_dev/community) or [open an issue](../../issues) so others
+can follow along and we'll support you as much as we can. When you're finished submit a
+pull request to the master branch referencing the specific issue you addressed.
 
 ### Steps to Contribute
 
-1. Look for open issues or open one.
-2. Discuss the problem and or propose a solution.
-3. Fork it!
-4. Branch off of `master`: `git checkout master`
-5. Create your feature branch: `git checkout -b my-new-feature`
-6. Make changes
-7. Commit your changes: `git commit -m 'Add some feature'`
-8. Push to the branch: `git push origin my-new-feature`
-9. Submit a pull request. Make sure it is based off of the `master` branch when
+1. Look for open issues or open one
+1. Discuss the problem and or propose a solution
+1. Fork it! (and clone fork locally)
+1. Branch off of `master`: `git checkout master`
+1. [Setup development environment](#setup-development-environment)
+1. Create your feature branch: `git checkout -b my-new-feature`
+1. Make changes
+1. Commit your changes: `git commit -m 'Add some feature'`
+1. Don't forget to fix issues from `pre-commit` pipeline (either add changes made by hooks
+   or fix them manually in case of `flake8`)
+1. Push to the branch: `git push origin my-new-feature`
+1. Submit a pull request. Make sure it is based off of the `master` branch when
    submitting! :D
+
+## Setup development environment
+
+1. install `poetry` (only once per machine):\
+   `curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -`\
+   or [checkout installation instruction](https://python-poetry.org/docs/#installation)
```

Comment:
There is also the `conda install -c conda-forge poetry` from conda forge,  but it is just a suggestion as the 2 links that you have indicated seem enough.

Reply:
no, that's not the best way because in this case `poetry` will exist in the same environment with re rest of dependencies. This script installs system-wide instance of `poetry`, so it exists despite all the Python envs in a system, so it can manage them properly.

Still this solution would work